### PR TITLE
doc: samples: add descriptions to sample sections

### PR DESCRIPTION
**Pull request by greg-fer**
Added generic description to each sample section.
The description includes links to the related documentation,
general information about the SDK samples, and information
about filtering boards in the VSC extension.
NCSDK-23685.

### CODE REVIEW

**Review by wiba-nordic — APPROVED** [review]

**Review by peknis — APPROVED** [review]

File: doc/nrf/samples/net.rst [hunk 1]
```diff
@@ -3,6 +3,16 @@
 Networking samples
 ##################
 
+This section lists the available |NCS| samples for networking development.
+These samples use an nRF91 Series device over LTE or an nRF70 Series device over Wi-Fi® for communication and connection.
+Additionally, you might want to check the :ref:`cellular_samples` sample and the :ref:`wifi_samples`.
```
Comment (richabp) [inline doc/nrf/samples/net.rst:8]:
```suggestion
Additionally, you might want to check the :ref:`cellular_samples` and the :ref:`wifi_samples`.
```

File: doc/nrf/samples/suit.rst [hunk 2]
```diff
@@ -3,6 +3,14 @@
 SUIT samples
 ############
 
+This section lists available |NCS| samples for :ref:`Device Firmware Update using SUIT for nRF54H Series of System-on-Chip <ug_nrf54h20_suit_dfu>`.
```
Comment (richabp) [inline doc/nrf/samples/suit.rst:6]:
```suggestion
This section lists available |NCS| samples for :ref:`Device Firmware Update using SUIT for the nRF54H Series of System-on-Chip <ug_nrf54h20_suit_dfu>`.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/samples.rst [hunk 3]
```diff
@@ -11,6 +11,8 @@ Zephyr also provides a variety of :ref:`zephyr:samples-and-demos`, including ver
 These samples are a good starting point for understanding how to put together your own application.
 However, Zephyr samples and applications are not tested and verified to work with the |NCS| releases.
 
+.. samples_general_info_start
+
 General information about samples in the |NCS|
```
Comment (mia-ko) [inline doc/nrf/samples.rst:16]: Was the missing space between intro sentence and the first bullet point intentional?
It doesn't render that bad in this case, but we do state that the first bullet point line should have a blank line before it.

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: doc/nrf/samples.rst (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/samples.rst:16]: It is intentional to have this kind of formatting, like <dt><dd> in XML:
image

**Review by richabp — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by b-gent — APPROVED** [review]

**Pull request merged**